shell/platform/darwin/macos/framework/Source/FlutterEngine.mm

```diff
@@ -586,24 +579,10 @@ - (void)setSemanticsEnabled:(BOOL)enabled {
     return;
   }
   _semanticsEnabled = enabled;
-  // Remove the accessibility children from flutter view before reseting the bridge.
-  if (!_semanticsEnabled && self.viewController.viewLoaded) {
-    self.viewController.flutterView.accessibilityChildren = nil;
-  }
-  if (!_semanticsEnabled && _bridge) {
-    _bridge.reset();
-  } else if (_semanticsEnabled && !_bridge) {
-    _bridge = [self createAccessibilityBridge:self viewController:self.viewController];
-  }
+  [_viewController notifySemanticsEnabledChanged];
```

Comment:
Thoughts on calling the view controller's method `setSemanticsEnabled` and passing `enabled` as a parameter? (Apologies if this is a repeat comment, I thought I had left this comment before but I couldn't find it)

Reply:
Ah found it here: https://github.com/flutter/engine/pull/38855#discussion_r1081863524

Current approach is to have a single source of truth. Marking as resolved